src/databricks/labs/ucx/source_code/base.py

```diff
@@ -69,23 +69,46 @@ class Convention(Advice):
     """A suggestion for a better way to write the code."""
 
 
+# The default schema to use when the schema is not specified in a table reference
+# See: https://spark.apache.org/docs/3.0.0-preview/sql-ref-syntax-qry-select-usedb.html
+DEFAULT_SCHEMA = 'default'
+
+
 class Linter:
     @abstractmethod
-    def lint(self, code: str) -> Iterable[Advice]: ...
+    def lint(self, code: str, schema: str) -> Iterable[Advice]: ...
+
+    @property
+    @abstractmethod
+    def schema(self):
+        return None
 
 
 class Fixer:
     @abstractmethod
     def name(self) -> str: ...
 
     @abstractmethod
-    def apply(self, code: str) -> str: ...
```

Comment:
What's the use of a property that returns null?

Reply:
This is no longer present with next push